components/rank/index.tsx

```diff
@@ -0,0 +1,171 @@
+import styled from 'styled-components';
+import byteSize from 'byte-size';
+
+import { ArrowDown, ArrowUp } from '@components/icons';
+
+import device from '@utils/media_queries';
+import type { Data } from '@utils/dataType';
+
+const RankWrapper = styled.div`
+	font-size: 0.75rem;
+
+	@media ${device.tablet} {
+		font-size: 1rem;
+
+		& th {
+			&:first-child {
+				padding-left: 2rem;
+			}
+
+			&:last-child {
+				padding-right: 2rem;
+			}
+		}
+
+		& td {
+			&:first-child > span {
+				padding-left: 2rem;
+			}
+
+			&:last-child > span {
+				padding-right: 2rem;
+			}
+		}
+	}
+
+	& > table {
+		width: 100%;
+		border-collapse: collapse;
+	}
+
+	& th {
+		text-align: left;
+		padding-top: 1.75rem;
+		padding-bottom: 1.75rem;
+		background: #121212;
+		position: sticky;
+		top: 0;
+
+		&:first-child {
+			padding-left: 1rem;
+		}
+
+		&:last-child {
+			padding-right: 1rem;
+		}
+	}
+
+	& td {
+		&:first-child > span {
+			border-radius: 8px 0 0 8px;
+			padding-left: 1rem;
+		}
+
+		&:last-child > span {
+			border-radius: 0 8px 8px 0;
+			padding-right: 1rem;
+		}
+
+		& > span {
+			display: block;
+			background: #2c2c2c;
+			padding-top: 1rem;
+			padding-bottom: 1rem;
+			margin-bottom: 1rem;
+		}
+	}
+`;
+
+const formatWalletAddress = (wallet: string): string => {
+	const first5Chars = wallet.slice(0, 5);
+	const last5Chars = wallet.slice(-5);
+
+	return `${first5Chars}...${last5Chars}`;
+};
+
+const positionIndicator = (position: number): string => {
+	const indicator = Math.abs(position);
+	const cent = indicator % 100;
+	if (cent >= 10 && cent <= 20) return 'th';
```

Comment:
style nit: We usually prefer adding the brackets to these `if` cases that simply return so its easier for the code reader to follow the returns through indentation

I believe @arielmelendez called me out on this one before 😅

Reply:
Fixed in 697b5596c0cd6d7fa415ddcb06f8d1c913b899a1